File: leave-tracker-ui/src/api/Leave/LeaveService.js
import axios from 'axios'
import { API_URL} from '../../Constants'

class LeaveService {

    retrieveLeave(empid) {
        //console.log('executed service')
        return axios.get(`${API_URL}/LeaveTracker/v1/getLeavesByEmployeeId/${empid}`);
    }

    getCountOfPlannedLeaves(empid, leaveType) {
        //console.log('executed service')
        return axios.get(`${API_URL}/LeaveTracker/v1/getCountOfLeaves/${empid}/Planned Leave`);
    }
    
    getCountOfSickLeaves(empid, leaveType) {
        //console.log('executed service')
        return axios.get(`${API_URL}/LeaveTracker/v1/getCountOfLeaves/${empid}/Sick Leave`);
    }

    getTotalCountOfLeavesPerMonth(month) {
        //console.log('executed service')
        return axios.get(`${API_URL}/LeaveTracker/v1/getTotalCountOfLeavesPerMonth/${month}`);
    }

    addOrUpdateLeave(leave) {
        //console.log('executed service')
        return axios.post(`${API_URL}/LeaveTracker/v1/updateLeaveDetail`, leave);
    }
   
    deleteLeaveById(leaveId) {
        //console.log('executed service')
        return axios.delete(`${API_URL}/LeaveTracker/v1/deleteLeaveById/${leaveId}`);
    }

}

export default new LeaveService()

File: leaveTrackerApi/src/main/java/com/tcs/leaveTracker/leaveTrackerApi/service/LeaveService.java
package com.tcs.leaveTracker.leaveTrackerApi.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.tcs.leaveTracker.leaveTrackerApi.entity.Leave;
import com.tcs.leaveTracker.leaveTrackerApi.repository.LeaveDao;
@Service
public class LeaveService {
	
	@Autowired
	LeaveDao leaveDao;
	
	public List<Leave> getLeavesByEmployeeId(Long employeeId) {
		return leaveDao.findByEmployeeId(employeeId);
	}

	public String updateLeave(List<Leave> leaves) {
	    leaves.forEach(leave -> leaveDao.save(leave) );
		return "Success";
	}

	public  void deleteById(long leaveId) {
		 leaveDao.deleteById(leaveId);	 
	}
	
	
	public long getCountOfLeaves(Long employeeId, String leaveType) {
		return leaveDao.findByEmployeeIdAndLeaveType(employeeId, leaveType).size();
	}
	
	public long getTotalCountOfLeavesPerMonth(int month) {
		return leaveDao.fetchLeavIdByMonth(month).size();
	}
	
}
